lib/jekyll/filters.rb

```diff
@@ -389,7 +389,11 @@ def groupable?(element)
     private
     def item_property(item, property)
       if item.respond_to?(:to_liquid)
-        item.to_liquid[property.to_s]
+        res = item.to_liquid
+        property.to_s.split(".").each do |attr|
+          res = res[attr]
+        end
+        res
```

Comment:
additionally, what is `res`? Is there a better name we could give it? I'm also not a giant fan of abbreviations, and would prefer to see `property` used in place of `prop` instead.